src/Dofs/ConstraintHandler.jl

```diff
@@ -154,6 +154,52 @@ function _add!(ch::ConstraintHandler, dbc::Dirichlet, interpolation::Interpolati
     append!(ch.prescribed_dofs, constrained_dofs)
 end
 
+function _add!(ch::ConstraintHandler, dbc::Dirichlet, bcnodes::Set{Int}, interpolation::Interpolation, field_dim::Int, offset::Int)
+    if interpolation !== default_interpolation(getcelltype(ch.dh.grid))
+        warn("adding constraint to nodeset is not recommended for sub/super-parametric approximations.")
+    end
+
+    ncomps = length(dbc.components)
+    nnodes = getnnodes(ch.dh.grid)
+    interpol_points = ndofs_per_cell(ch.dh)
+    _celldofs = fill(0, ndofs_per_cell(ch.dh))
+    node_dofs = zeros(Int, ncomps, nnodes)
+    visited = BitVector(nnodes)
+    visited .= false
+    for (cellidx, cell) in enumerate(ch.dh.grid.cells)
+        celldofs!(_celldofs, ch.dh, cellidx) # update the dofs for this cell
+        for idx in 1:min(interpol_points, length(cell.nodes))
```

Comment:
Why the `min`? Don't we always want to loop over all the nodes in the cell?

Reply:
When `interpol_points < length(cell.nodes)`, this will an avoid out-of-bounds error in `_celldofs[noderange[c]]`. And when `interpol_points > length(cell.nodes)`, it will just do the right thing.

Reply:
So this is for a case with e.g. quadratic geometry and linear function interpolation?

Reply:
Yes.

Reply:
So any node `i` without dofs will just have 0 in its `node_dofs[:,i]`.

Reply:
But `interpol_points` is just the number of dofs in the cell, not the "number of nodes with dofs", which is what you want here, right?

Reply:
Oh yes that's a mistake, oops!

Reply:
`getnbasefunctions(interpolation)` should do the right thing.